cmd/query/app/flags.go

```diff
@@ -48,39 +48,37 @@ var tlsHTTPFlagsConfig = tlscfg.ServerFlagsConfig{
 	Prefix: "query.http",
 }
 
-// QueryOptionsStaticAssets contains configuration for handling static assets
-type QueryOptionsStaticAssets struct {
-	// Path is the path for the static assets for the UI (https://github.com/uber/jaeger-ui)
-	Path string `valid:"optional" mapstructure:"path"`
-	// LogAccess tells static handler to log access to static assets, useful in debugging
-	LogAccess bool `valid:"optional" mapstructure:"log_access"`
+type UI struct {
+	// ConfigFile is the path to a configuration file for the UI.
+	ConfigFile string `mapstructure:"config_file" valid:"optional"`
+	// AssetsPath is the path for the static assets for the UI (https://github.com/uber/jaeger-ui).
+	AssetsPath string `mapstructure:"assets_path" valid:"optional" `
+	// LogAccess tells static handler to log access to static assets, useful in debugging.
+	LogAccess bool `mapstructure:"log_access" valid:"optional"`
 }
 
-// QueryOptionsBase holds configuration for query service shared with jaeger(v2)
-type QueryOptionsBase struct {
-	// BasePath is the base path for all HTTP routes
-	BasePath string
-
-	StaticAssets QueryOptionsStaticAssets `valid:"optional" mapstructure:"static_assets"`
-
-	// UIConfig is the path to a configuration file for the UI
-	UIConfig string `valid:"optional" mapstructure:"ui_config"`
-	// BearerTokenPropagation activate/deactivate bearer token propagation to storage
-	BearerTokenPropagation bool
-	// AdditionalHeaders
-	AdditionalHeaders http.Header
-	// MaxClockSkewAdjust is the maximum duration by which jaeger-query will adjust a span
-	MaxClockSkewAdjust time.Duration
-	// Tenancy configures tenancy for query
-	Tenancy tenancy.Options
+// OptionsBase holds configuration for query service shared with jaeger-v2
+type OptionsBase struct {
+	// BasePath is the base path for all HTTP routes.
+	BasePath string `mapstructure:"base_path"`
+	// UI contains configuration related to the Jaeger UI.
+	UI UI `mapstructure:"ui" valid:"optional"`
+	// BearerTokenPropagation activate/deactivate bearer token propagation to storage.
+	BearerTokenPropagation bool `mapstructure:"bearer_token_propagation"`
+	// MaxClockSkewAdjust is the maximum duration by which jaeger-query will adjust a span.
+	MaxClockSkewAdjust time.Duration `mapstructure:"max_clock_skew_adjust"`
 	// EnableTracing determines whether traces will be emitted by jaeger-query.
-	EnableTracing bool
+	EnableTracing bool `mapstructure:"enable_tracing"`
 }
 
-// QueryOptions holds configuration for query service
-type QueryOptions struct {
-	QueryOptionsBase
+// Options holds configuration for query service
+type Options struct {
+	OptionsBase
 
+	// Tenancy configures tenancy for query
+	Tenancy tenancy.Options
+	// AdditionalHeaders
+	AdditionalHeaders http.Header
```

Comment:
I don't like this. The objective should be to align v1 and v2 configs more, but you're going in the opposite direction - repeating more things in v2 which are done differently from v1. So far the only difference in v2 config were the storage names (which are legitimately different from v1 as a mechanism), and the HTTP/GRPC settings because we wanted to reuse OTEL-standard structs for those. I don't recall why tenancy was separate in v2 and not part of the base. But the ideal situation would be that v2 config is just storage names + Base struct, nothing else.

Reply:
Ah okay got it - sorry about that! Let me move these back to the BaseConfig.